core/base-service/coalesce-badge.spec.js

```diff
@@ -0,0 +1,257 @@
+'use strict'
+
+const { expect } = require('chai')
+const { getShieldsIcon } = require('../../lib/logos')
+const coalesceBadge = require('./coalesce-badge')
+
+describe('coalesceBadge', function() {
+  describe('Label', function() {
+    it('uses the default label', function() {
+      expect(coalesceBadge({}, {}, { label: 'heyo' }).text).to.deep.equal([
+        'heyo',
+        'n/a',
+      ])
+    })
+
+    // This behavior isn't great and we might want to remove it.
+    it('uses the category as a default label', function() {
+      expect(coalesceBadge({}, {}, {}, { category: 'cat' }).text).to.deep.equal(
+        ['cat', 'n/a']
+      )
+    })
+
+    it('preserves an empty label', function() {
+      expect(
+        coalesceBadge({}, { label: '', message: '10k' }, {}).text
+      ).to.deep.equal(['', '10k'])
```

Comment:
Just an arbitrary comment that I'm surprised prettier aligned these three `expect` blocks in these three different ways 😄